pytheus/backends.py

```diff
@@ -22,15 +23,26 @@ def get(self) -> float:
         pass
 
 
-class InvalidBackendClassException(PytheusException):
-    def __init__(self, class_name: str):
-        super().__init__(
-            f"Invalid backend class '{class_name}', please use any of the following: "
-            f"{', '.join([cls.__name__ for cls in Backend.__subclasses__()])}"
+def _import_backend_class(full_import_path: str) -> Backend:
+    path_chunks = full_import_path.split(".")
+    path_chunks_length = len(path_chunks)
+    if path_chunks_length <= 1:  # Empty string or not full path to the class
+        raise InvalidBackendClassException(
+            "Backend class could not be imported. Full import path needs to be provided, e.g. "
+            "my_package.my_module.MyClass"
+        )
+    module_path = ".".join(path_chunks[:-1])
+    class_name = path_chunks[-1]
+    try:
+        module = importlib.import_module(module_path)
+    except ImportError as e:
+        raise InvalidBackendClassException(f"Module '{module_path}' could not be imported: {e}")
```

Comment:
might be better to just raise the `ImportError` here ? In the end if the user misstype it it's actually the error that would be correct.

For this logic specifically I was thinking something on the lines of this but maybe doesn't matter:
```python
    module, klass = full_import_path.rsplit('.', 1)
    module = importlib.import_module(module)
    if not hasattr(module, klass):
        raise ImportError(f'{klass} not found in {module}')
    cls = getattr(module, klass)
    return cls()
```

Reply:
A good spot to add the `InvalidBackendClassException` might be right before returning the class by doing a subclass check (as it seems the base Backend class will be inherited for default behaviour)

```python
if not issubclass(cls, Backend):
    raise InvalidBackendClassException
```

this would require the `Backend` class to be decorated with `@runtime_checkable` from the typing module.

Maybe the init could be avoided as well and just require to fulfill the protocol that a `config` member is available (needs double checking)

Reply:
There is almost a 1-to-1 mapping between all standard exceptions and custom exceptions. The idea of custom exceptions is to give the user some extra information about what went wrong, e.g. "Oh, something is wrong about the backend class I configured in this library ..." then he would read the error message with the import information and find out.

That is more straightforward than a plain `ImportError`. The user would still figure out it was the backend class from the message, but it would be a bit less obvious.

PS: Forgot about the `rsplit`, that's cleaner :+1:

Reply:
As for your second comment, that's a nice addition that it got lost when I refactored this to use `importlib`.

Reply:
6fa340e1907eee59498c5071736509f23d843083